www/js/rover-api.js

```diff
@@ -0,0 +1,98 @@
+/*----- MISC GLOBALS -----*/
+var roverDomain = ''; //Later, if the rover is not hosting the webpage, its address will go here
+var roverApiPath = '/api/v1/';
+
+/*----- HELPER FUNCTIONS -----*/
+function roverResource(resource) {
+	return roverDomain+roverApiPath+resource;
+}
+
+/*----- ROVER API FUNCTIONS -----*/
+function sendMotorCommand(command, pin, speed) {
+	$.post(roverResource('sendcommand'),
+		{
+			command: command,
+			pin: pin,
+			speed: Number(speed)
+		}, function (response) {
+			//writeToConsole(response);
+		});
+}
+
+function saveDesign() {
+	xml = Blockly.Xml.workspaceToDom(workspace);
+	xmlString = Blockly.Xml.domToText(xml);
+	$.post(roverResource('blockdiagrams'), {bdString: xmlString, designName: designName}, function(response){
+	}).error(function(){
+			writeToConsole("There was an error saving your design to the rover");
+	});
+}
+
+function refreshSavedBds() {
+	$.get(roverResource('blockdiagrams'), function(json){
+		if (!json.result.length){
+			$('#savedDesignsArea').text("There are no designs saved on this rover");
+		} else {
+			$('#savedDesignsArea').empty();
+			json.result.forEach(function(entry) {
+				$('#savedDesignsArea').append("<a href='#' class='button' style='margin:10px;' onclick='return loadDesign(\""+entry+"\")'>"+entry+"</a>");

```

Comment:
I'm not a huge fan of inlining dom like this. Its hard to maintain and cumbersome to read and interpret. Jquery has a fairly nice fluid syntax for this:

```
$(document.createElement('a')).addClass('button')
  .html(entry)
  .attr('href', '#')
  .css('margin', '10px')
  .appendTo($("savedDesignsArea"))
  .click(function() {
    return loadDesign(entry);
  });
```
This is definitely just a style suggestion though!